File: HotelsCalifornia.API/Models/Invoice.cs
/*
Invoice ID (PK)
Member ID (FK, optional)
Reservation ID (FK, one fee to many reservations)
Is Paid?

*/

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HotelsCalifornia.Models;

public class Invoice
{
    [Key]
    public int Id { get; set; }

    [ForeignKey("Member")]
    public int? MemberId { get; set; }
    
    [ForeignKey("Reservation")]
    public List<int> ReservationId { get; set; } = new();

    public bool IsPaid { get; set; } = false;
}

File: HotelsCalifornia.UI/src/types/Invoice.ts

export interface Invoice
{
    id: number,
    memberId: number,
    reservationId: number,
    isPaid: boolean
}
export interface NewInvoice
{
    memberId: number,
    reservationId: number,
    isPaid: boolean
}
export interface UpdateInvoice
{
    Id: number,
    IsPaid: boolean
}